src/main/java/kitchenpos/menu/domain/MenuProduct.java

```diff
@@ -0,0 +1,58 @@
+package kitchenpos.menu.domain;
+
+import kitchenpos.product.domain.Product;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+
+@Entity
+public class MenuProduct {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long seq;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "menu_id")
+    private Menu menu;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "product_id")
+    private Product product;
+
+    private long quantity;
+
+    protected MenuProduct() {
+    }
+
+    public MenuProduct(Product product, long quantity) {
+        this.product = product;
+        this.quantity = quantity;
+    }
+
+    public BigDecimal calculatePrice() {
+        return product.calculatePrice(quantity);
+    }
+
+    public void addMenu(Menu menu) {
+        this.menu = menu;
+        if(!menu.getMenuProducts().contains(this)) {
```

Comment:
```suggestion
        if(!menu.contains(this)) {
```
getter를 쓰게 되면 캡슐화가 깨집니다
객체 내부 상황을 외부로 노출시키게 됩니다